library/ssl_tls.c

```diff
@@ -8999,7 +9009,8 @@ int mbedtls_ssl_set_hostname( mbedtls_ssl_context *ssl, const char *hostname )
         if( ssl->hostname == NULL )
             return( MBEDTLS_ERR_SSL_ALLOC_FAILED );
 
-        mbedtls_platform_memcpy( ssl->hostname, hostname, hostname_len );
+        /* Not using more secure mbedtls_platform_memcpy as hostname is not that critical */
+        memcpy( ssl->hostname, hostname, hostname_len );
```

Comment:
Also transmitted in the clear (in initial handshakes).

Reply:
fixed